common/common.go

```diff
@@ -182,6 +182,9 @@ const (
 	AnnotationKeyAppInstance = "argocd.argoproj.io/tracking-id"
 	AnnotationInstallationID = "argocd.argoproj.io/installation-id"
 
+	// AnnotationKeyAppInstanceID is the Argo CD server URL that is managing this resource
+	AnnotationKeyAppInstanceID = "argocd.argoproj.io/instance-id"
```

Comment:
The naming is confusing. `AppInstanceID` sounds like an id for the application instance rather than ArgoCD server. I'd also use a more verbose value, like `argocd.argoproj.io/argocd-server-instance-id`.